pkg/resources/provisioning.cattle.io/v1/cluster/validator_test.go

```diff
@@ -131,3 +136,60 @@ func Test_isValidName(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateMachinePoolName(t *testing.T) {
+	tests := []struct {
+		name string
+		fail bool
+	}{
+		{
+			name: strings.Repeat("12345678", 8),
```

Comment:
Nit: I'd choose more descriptive names for subtests, and the value to test I'd make a separate field.